docs(dpg): add introduction to `RequestContext` (#28237)
Azure/azure-sdk-for-net: sdk/core/Azure.Core/README.md

The main shared concepts of Azure.Core (and so Azure SDK libraries using Azure.Core) include:

- Configuring service clients, e.g. configuring retries, logging (`ClientOptions`).

> - Customizing request (`RequestContext`).

Author: Done

More on client configuration in [client configuration samples](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Configuration.md)

### Customzing Request Using `RequestContext`

> Besides general configuration of _serice client_ through `ClientOptions`, you can customize each individual request sent by _service client_

Reviewer: ```suggestion
Besides general configuration of _service clients_ through `ClientOptions`, it is possible to customize each individual request sent by a _service client_
```

Reviewer 2: You can only do this in protocol methods on a service client, and in select APIs on Generation1 Convenience Clients, where RequestContext is exposed as a parameter.

I'd suggest changing this to "[...] you can customize the request send by the _service client_ using protocol methods or convenience APIs that expose `RequestContext` as a parameter."

Author: Done

using `RequestContext`.

> _Service clients_ have methods that can be used to call Azure services. We refer to these client methods _service methods_. You can
> pass `RequestContext` to each _service method_ as a parameter.

Reviewer: Suggest combining this sentence with the previous paragraph. 

```suggestion
Each _service client_ method involving a call to an Azure service accepts a `RequestContext` instance as a parameter.
```

Reviewer 2: We should remove this, because not all service client methods expose `RequestContext`.

Author: I move the sentence regarding _service methods_ back to section "Accessing HTTP Response Details Using ```Response<T>```".

```C# Snippet:SetRequestContext
Response response = await client.GetPetAsync("snoopy", new RequestContext()
{

>     ErrorOptions = ErrorOptions.Default

Reviewer 2: I think it's unrealistic that we'd create a new instance of `RequestContext` to set a property to its default value -- could we instead create a new RequestContext and call `AddClassifier(404, isError: false)` on it before passing it to the method?

Author: Replaced with a classifier 404 case.

});
```

> More on request customization in [RequestContext samples](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/RequestContext.md)

Reviewer 2: This doc is outdated and needs to be updated, specifically this section: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/ProtocolMethods.md#using-requestoptions-to-customize-behavior

Author: updated

Reviewer 2: Thank you!

More on long-running operations in [long-running operation samples](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/LongRunningOperations.md)

### Customzing Request Using `RequestContext`

> Besides general configuration of _serice clients_ through `ClientOptions`, it is possible to customize the requests sent by _service clients_

Reviewer 2: ```suggestion
Besides general configuration of _service clients_ through `ClientOptions`, it is possible to customize the requests sent by _service clients_
```
